nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -15,498 +15,203 @@
 import copy
 import csv
 import json
-import math
 import os
+import shutil
+import sys
 from collections import OrderedDict as od
 from datetime import datetime
-from typing import List
+from typing import Dict, List, Tuple
 
-import librosa
 import numpy as np
-import soundfile as sf
-import torch
 
-from nemo.collections.asr.metrics.wer import WER
-from nemo.collections.asr.metrics.wer_bpe import WERBPE
-from nemo.collections.asr.models import ClusteringDiarizer, EncDecCTCModel, EncDecCTCModelBPE, EncDecRNNTBPEModel
+from nemo.collections.asr.metrics.wer import word_error_rate
+from nemo.collections.asr.models import ClusteringDiarizer
 from nemo.collections.asr.parts.utils.speaker_utils import (
+    audio_rttm_map,
     get_uniqname_from_filepath,
     labels_to_rttmfile,
     rttm_to_labels,
     write_rttm2manifest,
 )
-from nemo.collections.asr.parts.utils.streaming_utils import AudioFeatureIterator, FrameBatchASR
-from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
 from nemo.utils import logging
+from nemo.utils.decorators.experimental import experimental
 
-__all__ = ['ASR_DIAR_OFFLINE']
+try:
+    import arpa
+except ImportError:
+    logging.info("arpa is not installed. You must install arpa to refine diarization result with LM.")
+
+try:
+    import diff_match_patch
+except ImportError:
+    logging.info(
+        "diff_match_patch is not installed. You must install diff_match_patch to use CTM file based evaluation."
+    )
```

Comment:
Can we move this to case only when users try to use those features but not for every type of run?